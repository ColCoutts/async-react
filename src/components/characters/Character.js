import React from 'react';
import PropTypes from 'prop-types';

function Character({ character }) {
  const {
    name,
    gender,
    age,
    films
  } = character;

  return (
    <section>
      <p>{name}</p>
      <p>{gender}</p>
      <p>{age}</p>
      <p>{films}</p>
    </section>
  );
}

Character.propTypes = {
  character: PropTypes.shape({
    name: PropTypes.string.isRequired,
    gender: PropTypes.string.isRequired,
    age: PropTypes.string.isRequired,
    films: PropTypes.array.isRequired
  }).isRequired
};

export default Character;
