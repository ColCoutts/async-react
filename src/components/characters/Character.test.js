import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders a snapshot', () => {
    const character = {
      name: 'Colin',
      gender: 'Alient',
      age: '-less',
      films: [
        'this film'
      ]
    };

    const wrapper = shallow(<Character character={character} />);
    expect(wrapper).toMatchSnapshot();
  });
});
