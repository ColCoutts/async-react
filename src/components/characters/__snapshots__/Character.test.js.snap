// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders a snapshot 1`] = `
<section>
  <p>
    Colin
  </p>
  <p>
    Alient
  </p>
  <p>
    -less
  </p>
  <p>
    this film
  </p>
</section>
`;
